### PR TITLE
Metrics Glossary: TP / FP / FN / TN

### PR DESCRIPTION
**Pull request by y27choi**
### Linked issue(s):
[KOL-2540](https://linear.app/kolena/issue/KOL-2540/port-tpfpfntn-doc-to-docskolenaio)
[KOL-2541](https://linear.app/kolena/issue/KOL-2541/port-confusion-matrix-doc-to-docskolenaio)

### What change does this PR introduce and why?
Added TP/FP/FN/TN and confusion matrix doc page (without any style changes)

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

**Comment by gordonhart** [comment]
I've edited the TP / FP / FN / TN guide and removed the confusion matrix guide to keep the scope down. Ready to take a look @y27choi

File: docs/metrics/tp-fp-fn-tn.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+Rather than being computed at the sample level (e.g. per image), they're computed at the instance level (i.e. per object)
+for instances that the model is detecting. When given an image with multiple objects, each inference and each ground truth
+is assigned to one group, and the definitions of the terms are slightly altered:
+
+| Metric | | Description |
+| --- | --- | --- |
+| True Positive | TP | Inference matched with a ground truth |
+| False Positive | FP | Inference that is not matched with a ground truth |
+| <nobr>False Negative</nobr> | FN | Ground truth that is not matched with an inference |
+| True Negative | TN | <p>:kolena-warning-sign-16: **Poorly defined for object detection!** :kolena-warning-sign-16:</p><p>In object detection tasks, a true negative is any non-object that isn't detected as an object. This isn't well defined and as such true negative isn't a commonly used metric in object detection.</p><div>Occasionally, for object detection tasks "true negative" is used to refer to any image that does not have any true positive or false positive inferences.</div> |
```
Comment (y27choi) [inline docs/metrics/tp-fp-fn-tn.md:121]: `kolena-warning-sign` doesn't render for me when I built it locally.

image

Comment (gordonhart) [reply docs/metrics/tp-fp-fn-tn.md:121]: Have you restarted your docs server?

File: docs/metrics/tp-fp-fn-tn.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+a single score that summarizes model performance across all classes, there are a few different ways to aggregate
+per-class metrics: **macro**, **micro**, and **weighted.**
+
+Read more about these different averaging methods in the [Averaging Methods guide](./averaging-methods.md).
+
+## Limitations and Biases
+
+TP, FP, TN, and FN are four metrics based on the assumption that each sample/instance can be classified as a positive
+or a negative, thus they can only be applied to single-class applications. The workaround for multiple-class
+applications is to compute these metrics for each label using the **one-versus-rest** (OvR) strategy and then treat it
```
Comment (y27choi) [inline docs/metrics/tp-fp-fn-tn.md]: "**one-vs-rest** (OvR)" used [here](https://github.com/kolenaIO/kolena/pull/91/files#diff-62887e7f8c765524b6957bd5f04cd5ed5328015562d7b14db39e5186872678deR81)

Should we use "vs" or "versus"?

File: docs/metrics/tp-fp-fn-tn.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    | TP | FP | FN |
+    | --- | --- | --- |
+    | 1 | 1 | 1 |
+
+#### Multiclass
+
+Like classification, multiclass object detection tasks compute TP / FP / FN per class.
+
+??? example "Example: Multiclass Object Detection"
```
Comment (y27choi) [inline docs/metrics/tp-fp-fn-tn.md:164]: I love these collapsible example sections.

Are the legends too big?  I could make it smaller so it fits well with the image (or just make it part of each image).
Should we center the table?  IMO, it looks better when it's centered.

File: docs/metrics/tp-fp-fn-tn.md (hunk 1 above)
Comment (y27choi) [reply docs/metrics/tp-fp-fn-tn.md:121]: No, I haven't!  It works now 👍 
image

**Review by y27choi — COMMENTED** [review]
Left nit comments.  It looks really good.  Thanks for making this doc look so pretty, @gordonhart 👍 Approved!

File: docs/metrics/tp-fp-fn-tn.md (hunk 2 above)
Comment (gordonhart) [reply docs/metrics/tp-fp-fn-tn.md]: Updating to "vs"

File: docs/metrics/tp-fp-fn-tn.md (hunk 3 above)
Comment (y27choi) [reply docs/metrics/tp-fp-fn-tn.md:164]: What do you think of this? 👍  or 👎 ?
metrics-tpfpfntn-single-class
metrics-tpfpfntn-multi-class

File: docs/metrics/tp-fp-fn-tn.md
Comment (y27choi) [inline docs/metrics/tp-fp-fn-tn.md:1]: Should we add?

```
---
search:
  exclude: true
---
```

File: docs/metrics/tp-fp-fn-tn.md (hunk 3 above)
Comment (gordonhart) [reply docs/metrics/tp-fp-fn-tn.md:164]: In order to get these diagrams to look good, we need to keep a fair number of things in mind:

- Use consistent colors across diagrams: green=good, red=bad, things that are neither good or bad should be neither green nor red
- Consistent font family, color, and sizing
- Consistent application of things like stroke width and corner radius
- Different dark mode and light mode colors
- Consistent width — all images should be the same width such that there are no surprises when some images are stretched or squished to fit the main content column

It'll take some time to work out the flow for creating these graphics, but once it's worked out it should be low-lift to add more.

CC @pvanburen

**Review by gordonhart — APPROVED** [review]

File: docs/metrics/tp-fp-fn-tn.md
Comment (gordonhart) [reply docs/metrics/tp-fp-fn-tn.md:1]: Yes, adding. Good catch

File: docs/metrics/tp-fp-fn-tn.md (hunk 3 above)
Comment (gordonhart) [reply docs/metrics/tp-fp-fn-tn.md:164]: > What do you think of this? 

Better, but will still need to be updated to follow the above. Not sure if it's worth changing the image before landing this, considering that we'll make an editing pass through all of these metrics glossary docs to update diagrams sometime over the next few days.

Comment (y27choi) [reply docs/metrics/tp-fp-fn-tn.md:164]: Yup that makes sense

**Pull request merged**